packages/neuron-wallet/src/utils/buffer.ts

```diff
@@ -22,4 +22,12 @@ export default class BufferUtils {
     const buf = Buffer.from(leHex.slice(2), 'hex')
     return (buf.readBigUInt64LE(8) << BigInt(64)) + buf.readBigUInt64LE(0)
   }
+
+  public static parseAmountFromSUDTData(leHex: string): bigint {
+    try {
+      return this.readBigUInt128LE(leHex)
```

Comment:
```suggestion
      return this.readBigUInt128LE(leHex.slice(0, 34))
```

sUDT only read first 16 bytes of cell data

Reply:
Thanks @classicalliu for reviewing.

I think we want to follow the strict sUDT protocol standard for the data field here. If it does not strictly equal to 16 bytes, the `readBigUInt128LE ` will throw an error. Then `parseAmountFromSUDTData ` will catch and return `BigInt(0)` for this invalid sUDT data, so that the downstream won't crash or sum up the amount.